File: scripts/convert-crops.js
import fs from 'fs';
import path from 'path';
import sharp from 'sharp';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const cropsDir = path.resolve(__dirname, '../public/crops');

async function convertImages() {
  const files = fs.readdirSync(cropsDir);
  const pngFiles = files.filter(f => f.toLowerCase().endsWith('.png'));
  
  if (pngFiles.length === 0) {
    console.log('No PNG files found.');
    return;
  }

  let convertedCount = 0;
  let totalSavedBytes = 0;

  for (const file of pngFiles) {
    const inputPath = path.join(cropsDir, file);
    const outputPath = path.join(cropsDir, file.replace(/\.png$/i, '.webp'));
    
    try {
      const stats = fs.statSync(inputPath);
      const originalSize = stats.size;

      // Convert to WebP
      await sharp(inputPath)
        .webp({ quality: 80, effort: 4 })
        .toFile(outputPath);

      const newStats = fs.statSync(outputPath);
      const newSize = newStats.size;

      const saved = originalSize - newSize;
      totalSavedBytes += saved;
      convertedCount++;

      console.log(`Converted ${file} to WebP. Saved ${(saved / 1024).toFixed(2)} KB`);

      // Optionally delete original PNG to save space
      fs.unlinkSync(inputPath);
    } catch (err) {
      console.error(`Error converting ${file}:`, err);
    }
  }

  console.log(`\nSuccessfully converted ${convertedCount} images.`);
  console.log(`Total space saved: ${(totalSavedBytes / 1024 / 1024).toFixed(2)} MB`);
}

convertImages();

File: scripts/convert-crops.py
import os
from PIL import Image

crops_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../public/crops'))

if not os.path.exists(crops_dir):
    print(f"Directory {crops_dir} does not exist!")
    exit(1)

count = 0
saved_bytes = 0

for filename in os.listdir(crops_dir):
    if filename.lower().endswith('.png'):
        png_path = os.path.join(crops_dir, filename)
        webp_path = os.path.join(crops_dir, os.path.splitext(filename)[0] + '.webp')
        
        try:
            old_size = os.path.getsize(png_path)
            
            with Image.open(png_path) as img:
                img.save(webp_path, 'webp', quality=80)
            
            new_size = os.path.getsize(webp_path)
            saved = old_size - new_size
            saved_bytes += saved
            count += 1
            
            print(f"Converted {filename} to WebP. Saved {saved / 1024:.2f} KB")
            
            os.remove(png_path)
        except Exception as e:
            print(f"Failed to convert {filename}: {e}")

print(f"\nSuccessfully converted {count} images.")
print(f"Total space saved: {saved_bytes / 1024 / 1024:.2f} MB")
